plutus-core/testlib/PlutusCore/Generators/PIR/Builtin.hs

```diff
@@ -0,0 +1,144 @@
+{-# LANGUAGE DefaultSignatures #-}
+{-# LANGUAGE GADTs             #-}
+{-# LANGUAGE PolyKinds         #-}
+{-# LANGUAGE TypeApplications  #-}
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module PlutusCore.Generators.PIR.Builtin where
+
+import Data.ByteString (ByteString)
+import Data.Coerce
+import Data.Kind qualified as GHC
+import Data.Proxy
+import Data.Text (Text)
+import Data.Text qualified as Text
+import Data.Text.Encoding qualified as Text
+import Test.QuickCheck
+
+import PlutusCore
+import PlutusCore.Builtin
+import PlutusCore.Data
+
+instance Arbitrary Data where
+    arbitrary = error "implement me"
+    shrink = error "implement me"
+
+-- | Same as 'Arbitrary' but specifically for Plutus built-in types, so that we are not tied to
+-- the default implementation of the methods for a built-in type.
```

Comment:
okay, then I guess we should have justifications in each of the cases where we deviate from the existing defintion?

Reply:
Done for `Integer`. I don't know if there's any default handling of `Text` or `ByteString`.